Tests/PackageLoadingTests/PackageBuilderTests.swift

```diff
@@ -1659,6 +1659,30 @@ class PackageBuilderTests: XCTestCase {
         }
     }
 
+    func testUnknownSourceFilesUnderDeclaredSourcesIgnoredInV5_2Manifest() throws {
```

Comment:
Can you also write a unit test for 5.3 or do we already cover that using some other test?

Reply:
It's already covered, but maybe a more targeted one is a good idea.

Reply:
I've added a more targeted test.  I'm not sure that SwiftPM is doing the right thing in the case of missing rules, though... it doesn't seem logical to try to compile files with unknown suffixes.

Reply:
That's a separate issue, and shouldn't block this fix, but something to discuss.

Reply:
But that does seem to be the intentional 5.3 semantics in the source code.

Reply:
What was the reasoning for introducing the change in 5.3 to not validate source extensions? You worked on that @aciidb0mb3r right?

Reply:
From talking with Ankit it seems to have been just a bug.  Certainly anything that makes package manifests with an old tools version behave differently is by definition a bug.

Reply:
Right. This looks like a bug to me. I don't know if we actually want the other behavior in 5.3 (it's hopefully laid out in the proposal). It sounds like @abertelrud is planning to look at it separately.

Reply:
This is discussed in the "Rules for determining resource files" section of the proposal, but perhaps not in sufficient detail.  There were many iterations on the proposed semantics, but I thought where we ended up is that, while unambiguous types of files such as .metal, .storyboard, .xcassets, etc should be treated as resources, any other types would need to be specified as resources.  So this should result in an error requiring the user to either specify it as excluded or as a resource.

It sounds as if this needs to be clarified in a follow-on proposal, and the whole question of how to select rules for input files is of course related to the question of how to extend those rules for custom types (which is clearly beyond the scope of just clarifying the behavior here).

As things stand, what will happen if you mix in an unknown type with known compilable types (such as .c, .swift, etc) in 5.3 is that you'll get an error saying that multi-language targets aren't supported.  That's not ideal, but it's not a silent issue.  It's only in the case of an entire target full of unknown types of the same kind that it will try to compile them.

Reply:
Are we comfortable with the fix for the regression, and taking the question of 5.3 behavior as a follow-up discussion?

Reply:
I've been doing some more local testing of this, and will merge.

Reply:
Can you file a JIRA for tracking the issue for 5.3?

Reply:
https://bugs.swift.org/browse/SR-12667